layer.go

```diff
@@ -67,46 +69,49 @@ type Layer struct {
 	// finalHandler stores the final middleware chain handler.
 	finalHandler http.Handler
 
-	// memo stores the memoized middleware call chain by specific phase.
-	memo map[string]http.Handler
```

Comment:
Ok, I like this kind a reasoning e.g. benchmarking and taking a decision based on that.